great_expectations/experimental/datasources/interfaces.py

```diff
@@ -151,6 +154,7 @@ class DataAsset(ExperimentalBaseModel, Generic[_DatasourceT]):
 
     # non-field private attributes
     _datasource: _DatasourceT = pydantic.PrivateAttr()
+    _data_connector: DataConnector = pydantic.PrivateAttr()
```

Comment:
this field is not used and should not be here.

The fact that a data asset uses a data connector should not be part of the contract (as your own comment indicates, it is optional). If a data asset uses a data connector, it should initialize it in the constructor.